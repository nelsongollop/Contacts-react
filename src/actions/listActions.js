import axios from "axios";

export function getContacts () {
    return{
        type: "GET_CONTACTS",
        payload: new Promise((resolve, reject) => {
            axios.get('http://localhost:3001/api/v1/contacts')
                .then(
                    (result) => {
                        resolve(result.data)
                    },
                    (error) => {
                        reject(error)
                    }
                )
        })
    }
}

export function filter(keywords) {
    return {
        type: "FILTER_LIST",
        payload: keywords
    }
}

export function addContact(contact){
    return{
        type: "ADD_CONTACT",
        payload: contact
    }
}

