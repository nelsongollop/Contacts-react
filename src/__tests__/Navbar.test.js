import React from 'react';
import {Provider} from 'react-redux'
import configureStore from 'redux-mock-store';
import {configure, shallow, mount} from 'enzyme'
import Adapter from 'enzyme-adapter-react-16';
import Navbar from '../components/NavBar'
import {setUpdate, setSelected, filter} from "../actions/listActions";

configure({ adapter: new Adapter() });
const mockStore = configureStore([]);

describe('Navbar', () => {
    let store;
    let wrapper;
    beforeEach(() => {
        store = mockStore();
        store.dispatch = jest.fn();

        wrapper = mount(
            <Provider store={store}>
                <Navbar />
            </Provider>
        );
    });

    it('should render a Navbar', () => {
        expect(wrapper).toMatchSnapshot();
    });

    it('The button should setSelected and setUpdate', () => {
        wrapper.find('button').simulate('click')

        expect(store.dispatch).toHaveBeenCalledTimes(2);
        expect(store.dispatch).toHaveBeenCalledWith(
            setSelected(null)
        );
        expect(store.dispatch).toHaveBeenCalledWith(
            setUpdate(false)
        );
    });

    it('The textfield should trigger filter function', () => {
        let input = wrapper.find('input')
        input.simulate('change');

        expect(store.dispatch).toHaveBeenCalledTimes(1);
        expect(store.dispatch).toHaveBeenCalledWith(
            filter(input.text())
        );
    });
});
