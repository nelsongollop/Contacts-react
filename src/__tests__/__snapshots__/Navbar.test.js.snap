// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar should render a Navbar 1`] = `ReactWrapper {}`;
